src/components/DataDetails.vue

```diff
@@ -0,0 +1,274 @@
+    },
+    actionKeys: {
+      type: Array,
+      default: () => DefaultActionKeys,
+    },
+  },
+  data() {
+    return {
+      showUpsert: false,
+    };
+  },
+  inject: ['girderRest'],
+  asyncComputed: {
+    async details() {
+      // Use created as a litmus text for the existence of the rest of the data model.
+      // TODO: add proper model validation
+      if (this.datum && this.datum.created) {
+        return this.datum;
+      } else if (this.datum) {
+        const { data } = await this.girderRest.get(`${this.value[0]._modelType}/${this.value[0]._id}`);
+        return data;
+      }
+      return null;
+    },
+    files: {
+      default: [],
+      async get() {
+        if (this.datum && this.datum._modelType === 'item') {
+          const { data } = await this.girderRest.get(`item/${this.datum._id}/files`);
+          return data.map(f => f.name);
+        }
+        return [];
+      },
+    },
+  },
+  computed: {
+    title() {
+      return this.datum
+        ? (this.datum.name || this.formatUsername(this.datum))
+        : `${this.value.length} Selections`;
+    },
+    datum() {
+      return this.value.length === 1 ? this.value[0] : undefined;
+    },
+    icon() {
+      return this.datum
+        ? this.$vuetify.icons.values[this.datum._modelType]
+        : this.$vuetify.icons.values.fileMultiple;
+    },
+    meta() {
+      if (this.details && 'meta' in this.details) {
+        return Object.entries(this.details.meta)
+          .map(([key, value]) => ({ key, value }));
+      }
+      return [];
+    },
+    info() {
+      if (this.details) {
+        /* If this is a single datum */
+        return this.infoKeys.map((k) => {
+          let val = this.details[k.value];
+          if (k.transform) {
+            val = k.transform(val);
+          }
+          return `${k.name}${val}`;
+        });
+      } else if (this.value.length > 1) {
+        /* If this is a multi-selection */
+        const reducer = (acc, curr) => {
+          acc[curr._modelType] += 1;
+          acc.size += curr.size;
+          return acc;
+        };
+        const typeCounts = this.value.reduce(reducer, {
+          item: 0,
+          folder: 0,
+          size: 0,
+        });
+        const countMessages = ['item', 'folder']
+          .filter(k => typeCounts[k] > 0)
+          .map(k => `${typeCounts[k]} ${k}(s) selected`);
+        const sizeMessage = `Total size: ${this.formatSize(typeCounts.size)}`;
+        return [...countMessages, sizeMessage];
+      }
+      return [];
+    },
+    actions() {
+      const actionType = this.datum ? this.datum._modelType : 'multi';
+      return this.actionKeys.filter(k => k.for.includes(actionType));
+    },
+  },
+  methods: {
+    async handleAction(action) {
+      await action.handler.apply(this);
+      this.$emit('action', action);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+.data-details {
+  width: 100%;
+
+  .details-title {
+    font-size: 16px;
+  }
+}
```

Comment:
Why is `width: 100%` necessary?

Also, could we use one of the semantic classes from vuetify rather than hard-coding a font-size?

Reply:
Done.  Also added a kinda trivial test as little more than a sanity check. I'll add some more later.

Reply:
Going from nothing to a "does it explode or not" test has a great ROI. Good stuff.